### PR TITLE
Invert tree python implementation

### PR DESCRIPTION
**Pull request by charitymarani**
## What does this PR do?
  
- Creates a tree from a list of node values
- Inverts the tree
- Traverses the tree In-Order

## How to test
- Run `python samples/invert_tree_charity.py`
- This will print out the nodes In-order before Inversion and after Inversion

### CODE REVIEW

File: samples/invert_tree_charity.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        # if data is smaller than parent , insert it into left side
+        if data < node.data:
+            node.left = self.insert(node.left, data)
+        elif data > node.data:
+            node.right = self.insert(node.right, data)
+
+        return node
+
+
+    def traverse(self, root):
```
Comment (profnandaa) [inline samples/invert_tree_charity.py]: nit: is there need to supply `root` when it's already in `self.root`?

File: samples/invert_tree_charity.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        swap left and right nodes
+        """
+        node.left,node.right = node.right,node.left
+
+
+
+    
+def main():
+    arr = [10, 5, 20, 4, 3, 12 ]
+    tree = Tree(arr)
```
Comment (profnandaa) [inline samples/invert_tree_charity.py]: Though later I noticed that perhaps was not a good idea to have the Tree initialized by an array. I made some changes to our initial class code, you can have a look at it.

**Review by profnandaa — APPROVED** [review]
LGTM generally.

File: samples/invert_tree_charity.py (hunk 1 above)
Comment (charitymarani) [reply samples/invert_tree_charity.py]: I've created a helper function to help with this

File: samples/invert_tree_charity.py (hunk 2 above)
Comment (charitymarani) [reply samples/invert_tree_charity.py]: I've updated the implementation

**Pull request merged**